user-service/src/main/java/com/ecommercepractice/userservice/exception/ControllerAdvisor.java

```diff
@@ -0,0 +1,76 @@
+package com.ecommercepractice.userservice.exception;
+
+import org.json.JSONObject;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class ControllerAdvisor extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(UserAlreadyCreatedException.class)
+    public ResponseEntity<Object> handleUserAlreadyCreatedException(
+            UserAlreadyCreatedException ex, WebRequest request){
+
+        JSONObject body = new JSONObject();
```

Comment:
Update JSONObjet to you own class for error details

Reply:
Done